plugins/modules/dcnm_image_policy.py

```diff
@@ -0,0 +1,858 @@
+
+        # Change "N9K/N3K" to "N9K" in have to match the request payload.
+        if have.get("platform", None) == "N9K/N3K":
+            have["platform"] = "N9K"
+
+        # If keys are not set in both have and want, remove them.
+        for key in ["agnostic", "epldImgName", "packageName", "rpmimages"]:
+            if have.get(key, None) is None and want.get(key, None) is None:
+                have.pop(key, None)
+                want.pop(key, None)
+
+            if have.get(key, None) == "" and want.get(key, None) == "":
+                have.pop(key, None)
+                want.pop(key, None)
+        return (have, want)
+
+    def _merge_policies(self, have: Dict, want: Dict) -> Dict:
+        """
+        Merge the parameters in want with the parameters in have.
+
+        Caller: self.commit()
+        """
+        (have, want) = self._prepare_for_merge(have, want)
+
+        # Merge the parameters in want with the parameters in have.
+        # The parameters in want take precedence.
+        merge = MergeDicts(self.ansible_module)
+        merge.dict1 = have
+        merge.dict2 = want
+        merge.commit()
+        merged = copy.deepcopy(merge.dict_merged)
+
+        needs_update = False
+
+        if have != merged:
+            needs_update = True
+
+        return (merged, needs_update)
+
+    def send_need_create(self) -> None:
+        """
+        Create the policies in self.need_create
+
+        Callers:
+        - self.handle_merged_state()
+        """
+        self.image_policy_create.results = self.results
+        self.image_policy_create.payloads = self.need_create
+        self.image_policy_create.commit()
+
+    def send_need_update(self) -> None:
+        """
+        Update the policies in self.need_update
+
+        Callers:
+        - self.handle_merged_state()
+        """
+        self.image_policy_update.results = self.results
+        self.image_policy_update.payloads = self.need_update
+        self.image_policy_update.commit()
+
+
+def main():
+    """
+    main entry point for module execution
+    """
+
+    element_spec = {
+        "config": {
+            "required": False,
+            "type": "list",
+            "elements": "dict",
+            "default": [],
+        },
+        "state": {
+            "default": "merged",
+            "choices": ["deleted", "merged", "overridden", "query", "replaced"],
+        },
+    }
+    ansible_module = AnsibleModule(argument_spec=element_spec, supports_check_mode=True)
+
+    # Create the base/parent logger for the dcnm collection.
+    # To enable logging, set enable_logging to True.
+    # log.config can be either a dictionary, or a path to a JSON file
+    # Both dictionary and JSON file formats must be conformant with
+    # logging.config.dictConfig and must not log to the console.
+    # For an example configuration, see:
+    # $ANSIBLE_COLLECTIONS_PATH/cisco/dcnm/plugins/module_utils/common/logging_config.json
+    enable_logging = False
```

Comment:
This should be changed to use environment variable as was done in dcnm_fabric.py.  For now, since enable_logging is False, it'll be fine.  I've made a reminder for myself to fix this later.